test/utils/test_http_client.py

```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import pytest
+from haystack.utils.http_client import init_http_client
+
+
+def test_init_http_client():
+    http_client = init_http_client(http_client_kwargs={"base_url": "https://example.com"})
+    assert http_client is not None
+    assert http_client.base_url == "https://example.com"
+
+
```

Comment:
could you please also test other cases?

- `init_http_client()` (no parameter)
- `init_http_client(async_client=True)`

You can find a similar tests here
https://github.com/deepset-ai/haystack/blob/656fe6dc6edc9d4a079b9e44330dd2a943aad77b/test/components/embedders/test_azure_text_embedder.py#L176-L185